File: php/features.php
<?php
declare(strict_types=1);

// ExtinctAnimals SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class ExtinctAnimalsFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new ExtinctAnimalsBaseFeature();
            case "test":
                return new ExtinctAnimalsTestFeature();
            default:
                return new ExtinctAnimalsBaseFeature();
        }
    }
}

File: py/extinctanimals_sdk/features.py
# ExtinctAnimals SDK feature factory

from extinctanimals_sdk.feature.base_feature import ExtinctAnimalsBaseFeature
from extinctanimals_sdk.feature.test_feature import ExtinctAnimalsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ExtinctAnimalsBaseFeature(),
        "test": lambda: ExtinctAnimalsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
